components/engine_traits/src/engine.rs

```diff
@@ -59,6 +59,11 @@ pub trait KvEngine:
     /// This only exists as a temporary hack during refactoring.
     /// It cannot be used forever.
     fn bad_downcast<T: 'static>(&self) -> &T;
+
+    /// Can apply snapshot now
+    fn can_apply_snapshot(&self) -> bool {
```

Comment:
when should can_apply_snapshot return false

Reply:
Agree, better to add comment for when it will return false.